apps/meteor/tests/end-to-end/api/rooms.ts

```diff
@@ -1229,8 +1229,138 @@ describe('[Rooms]', () => {
+				testDiscussionOutsideTeam = (
+					await request
+						.post(api('rooms.createDiscussion'))
+						.set(credentials)
+						.send({
+							prid: testChannelOutside._id,
+							t_name: `test-discussion-${Date.now()}`,
+						})
+				).body.discussion;
+			});
+
+			after(() =>
+				Promise.all([
+					deleteRoom({ type: 'c', roomId: testChannel._id }),
+					deleteRoom({ type: 'p', roomId: testDiscussion._id }),
+					deleteRoom({ type: 'c', roomId: testChannelOutside._id }),
+					deleteRoom({ type: 'p', roomId: testDiscussionOutsideTeam._id }),
+					deleteTeam(credentials, teamName),
+				]),
+			);
+
+			it('should return the channel info, team and parent info', async () => {
+				const result = await request.get(api('rooms.info')).set(credentials).query({ roomId: testChannel._id }).expect(200);
+
+				expect(result.body).to.have.property('success', true);
+				expect(result.body).to.have.property('team');
+				expect(result.body).to.have.property('parent');
+				expect(result.body.parent).to.have.property('_id').and.to.equal(testTeam.roomId);
+			});
+
+			it('should return the dicsussion room info and parent info', async () => {
+				await request
+					.get(api('rooms.info'))
+					.set(credentials)
+					.query({ roomId: testDiscussion._id })
+					.expect(200)
+					.expect((res) => {
+						expect(res.body).to.have.property('success', true);
+						expect(res.body).to.have.property('parent').and.to.be.an('object');
+						expect(res.body.parent).to.have.property('_id').and.to.be.equal(testChannel._id);
+					});
+			});
+
+			it('should not return parent info for the main room of the team', async () => {
+				await request
+					.get(api('rooms.info'))
+					.set(credentials)
+					.query({ roomId: testTeam.roomId })
+					.expect(200)
+					.expect((res) => {
+						expect(res.body).to.have.property('success', true);
+						expect(res.body).to.not.have.property('parent');
+						expect(res.body).to.have.property('team');
+					});
+			});
+
+			it('should not return team for room outside team', async () => {
+				await request
+					.get(api('rooms.info'))
+					.set(credentials)
+					.query({ roomId: testChannelOutside._id })
+					.expect(200)
+					.expect((res) => {
+						expect(res.body).to.have.property('success', true);
+						expect(res.body).to.not.have.property('team');
+						expect(res.body).to.not.have.property('parent');
+					});
+			});
+
+			it('should return the parent for discussion outside team', async () => {
+				await request
+					.get(api('rooms.info'))
+					.set(credentials)
+					.query({ roomId: testDiscussionOutsideTeam._id })
+					.expect(200)
+					.expect((res) => {
+						expect(res.body).to.have.property('success', true);
+						expect(res.body).to.have.property('parent').and.to.be.an('object');
+						expect(res.body.parent).to.have.property('_id').and.to.be.equal(testChannelOutside._id);
+						expect(res.body).to.not.have.property('team');
+					});
+			});
+		});
+	});
```

Comment:
Can we create a test case for a discussion created within the team(inside the team's main room)?